core/src/main/java/com/linecorp/armeria/server/RouteCache.java

```diff
@@ -122,7 +122,7 @@ private RouteCache() {}
         private final Function<V, Route> routeResolver;
         private final Cache<RoutingContext, V> findCache;
         private final Cache<RoutingContext, List<V>> findAllCache;
-        private final Set<Route> noCacheRoutes;
+        private final Set<Integer> noCacheRoutes;
```

Comment:
Even if `Route` did not override `hashCode()`, it is still possible that two different objects happen to have the same hash code, so I think you need to use something like `IdentityHashMap`.

Reply:
Ah, I see. I forgot `IdentityHashMap`. :-( Thanks!